qiskit/pulse/ir/ir.py

```diff
@@ -260,6 +256,31 @@ def edge_map(_x, _y, _node):
 
         return block
 
+    def copy(self) -> SequenceIR:
+        """Semi-deep copy of ``SequenceIR``.
+
+        The returned copy can be safely mutated without affecting the original object, while immutable
+        objects are still passed as reference for memory efficiency.
+
+        ``SequenceIR`` is poorly suited for both shallow and deep copy. A shallow copy
+        will contain references to mutable properties like ``sequence`` and ``time_table``.
+        A deep copy on the other hand will needlessly copy immutable objects like
+        :class:`.qiskit.pulse.Instruction`. This function returns a semi-deep copy -
+        A new object containing new objects for ``sequence`` and ``time_table``. However,
+        node data of type :class:`.qiskit.pulse.Instruction` will be passed as a reference.
+        Nested ``SequenceIR`` objects are copied using the same logic.
+
+        Returns: A semi-deep copy of the object.
+        """
+        copied = SequenceIR(self.alignment)
```

Comment:
```suggestion
        copied = self.__class__(self.alignment)
```
In case someone make a subclass in future.